op-node/node/runtime_config.go

```diff
@@ -78,3 +83,97 @@ func (r *RuntimeConfig) Load(ctx context.Context, l1Ref eth.L1BlockRef) error {
 	r.log.Info("loaded new runtime config values!", "p2p_seq_address", r.p2pBlockSignerAddr)
 	return nil
 }
+
+// Update the runtime config with any new L1 information from the given L1 block.
+// The update may be ignored if it's older than the current state.
+// If it's newer with a gap, or if there is a reorg, then it forces a full configuration reload.
+func (r *RuntimeConfig) Update(ctx context.Context, l1Ref eth.L1BlockRef) error {
+	if r.l1Ref.Hash == l1Ref.Hash {
+		r.log.Debug("skipping already processed L1 block", "skipped", l1Ref)
+		return nil
+	}
+	if r.l1Ref.Number < l1Ref.Number {
+		r.log.Debug("skipping update with older L1 block", "current", r.l1Ref, "skipped", l1Ref)
+		return nil
+	}
+	if r.l1Ref.Hash == l1Ref.ParentHash {
+		// apply update that fits on top
+		_, receipts, err := r.l1Client.FetchReceipts(ctx, l1Ref.Hash)
+		if err != nil {
+			return fmt.Errorf("failed to fetch receipts to update runtime config with latest changes of %s: %w", l1Ref, err)
+		}
+
+		runtimeConfigCopy := r.runtimeConfigData
+		if err := parseReceipts(r.rollupCfg, receipts, &runtimeConfigCopy); err != nil {
+			return fmt.Errorf("failed to parse receipts to update runtime config with latest changes of %s: %w", l1Ref, err)
+		}
+
+		// concurrent updates / races are fine, as long as the data is consistent with its reference of origin in L1
+		r.mu.Lock()
+		r.l1Ref = l1Ref
+		r.runtimeConfigData = runtimeConfigCopy
+		r.mu.Unlock()
+
+		return nil
+	} else {
+		// fully reload
+		r.log.Warn("skipped or reorged on runtime config, reloading now", "old", r.l1Ref, "new", l1Ref)
+		return r.Load(ctx, l1Ref)
+	}
+}
+
+// parseReceipts iterates receipts of the SystemConfig to find runtime configuration updates
+func parseReceipts(cfg *rollup.Config, receipts types.Receipts, dest *runtimeConfigData) error {
+	var result error
+	for i, rec := range receipts {
+		if rec.Status != types.ReceiptStatusSuccessful {
+			continue
+		}
+		for j, log := range rec.Logs {
+			if log.Address == cfg.L1SystemConfigAddress && len(log.Topics) > 0 && log.Topics[0] == derive.ConfigUpdateEventABIHash {
```

Comment:
slight preference for reducing the size of this `if` statement with a helper function, something like `isConfigUpdate(log *types.Log) bool`